sdk/python/kfp/components/pipeline_task.py

```diff
@@ -474,6 +475,42 @@ def my_pipeline():
             self._task_spec.dependent_tasks.append(task.name)
         return self
 
+    def ignore_upstream_failure(self) -> 'PipelineTask':
+        """If called, the pipeline task will run when any specified upstream
+        tasks complete, even if unsuccessful.
+
+        This method effectively turns the caller task into an exit task
+        if the caller task has upstream dependencies.
+
+        If the task has no upstream tasks, either via data exchange or an explicit dependency via .after(), this method has no effect.
+
+        Returns:
+            Self return to allow chained setting calls.
+
+        Example:
+          ::
+
+            @dsl.pipeline()
+            def my_pipeline(text: str = 'message'):
+                task = fail_op(message=text)
+                clean_up_task = print_op(
+                    message=task.output).ignore_upstream_failure()
+        """
+
+        for input_spec_name, input_spec in (self.component_spec.inputs or
+                                            {}).items():
+            argument_value = self._inputs[input_spec_name]
+            if (isinstance(argument_value, pipeline_channel.PipelineChannel)
+               ) and (not input_spec.optional) and (argument_value.task_name
+                                                    is not None):
+                raise ValueError(
+                    f'Tasks can only use .ignore_upstream_failure() if all input parameters that accept arguments created by an upstream task have a default value, \n\tin case the upstream task fails to produce its output. Input parameter task {self.name}`s {input_spec_name!r} argument is an output of an upstream task {argument_value.task_name}, but {input_spec_name} has no default value.'
```

Comment:
nit: if going to use `!r`, prefer using for all of the interpolated strings that represent user code symbols/names

Reply:
Separately, what is `\n\t` for here?

Reply:
> Separately, what is \n\t for here?

It gives a cleaner print of the error on the terminal

Reply:
I think that might depend on the width of the terminal. What happens if you make your terminal more narrow? This is what mine looks like:

<img width="470" alt="image" src="https://user-images.githubusercontent.com/55268212/218547114-12d3a8c5-1193-44a7-ac64-6616d03a47fe.png">

Reply:
Ok! Ill just remove the tag